mne_bids/meg_bids.py

```diff
@@ -117,8 +125,20 @@ def _scans_tsv(raw, raw_fname, fname, verbose):
         acq_time = datetime.fromtimestamp(
             meas_date).strftime('%Y-%m-%dT%H:%M:%S')
 
-    df = pd.DataFrame({'filename': ['%s' % raw_fname],
-                       'acq_time': [acq_time]})
+    # check to see if the file already exists.
```

Comment:
why do we not have an overwrite for `_scans_tsv` as we do for `_events_tsv`? it seems that it will append and not just overwrite

Reply:
I wasn't exactly sure on the best thing to do here.
My reasoning was this:
If overwrite is True we want the files to be overwritten (obviously). However, if we have multiple acquisitions or tasks in the same folder, they all need to be listed in the same scans.tsv file, so we require them to append the current data if it isn't in there currently.
Therefore overwrite = True for other files has a different intended effect to what overwrite = True would have on the scans.tsv (ie. create a new scans.tsv file)

Reply:
hmm ... okay good point. So, we should basically overwrite `scans.tsv` when `overwrite=True` but generate it in the end. It should check the BIDS folder for raw files and use it to generate the list in `scans.tsv`. That way, you don't care if raw files have been added or removed. It will always accurately reflect what is contained in the folder. Wdyt @monkeyman192 @teonbrooks ?

Reply:
I feel like having to check what raw files are in a folder could lead to a lot more problems. For one you will need to know what files are considered raw files which seems like more effort.
Isn't how it works currently essentially that anyway? Any raw files currently in the folder will be listed, and if you add another it will be added. If you want to start again you would need to clear out the whole folder which goes back to the folder overwrite issue.
Also, side note: the path in the scans.tsv isn't actually correct (it should be a path relative to the scans.tsv file).

Reply:
> I feel like having to check what raw files are in a folder could lead to a lot more problems. For one you will need to know what files are considered raw files which seems like more effort.

you do know if a file is a raw file just from the filename, no? Derivatives are stored in a separate folder for BIDS and there is an ongoing discussion as to how to name them. Basically, any files that end with extension *.fif, ds/*, *.con etc ...

> Isn't how it works currently essentially that anyway? Any raw files currently in the folder will be listed, and if you add another it will be added.

nopes, not at the moment.

> If you want to start again you would need to clear out the whole folder which goes back to the folder overwrite issue.

hmm ... we need to find a clean solution for this. Later ...

> Also, side note: the path in the scans.tsv isn't actually correct (it should be a path relative to the scans.tsv file).

oh interesting, feel free to raise an issue or fix it in this PR

Reply:
> nopes, not at the moment.
How so? Every time you run the script any raw data will get added to the list, so unless you are adding raw files not with the script the scans.tsv file should have a list of every raw file in the directory.
> hmm ... we need to find a clean solution for this. Later ..
This is what I had proposed with the `overwrite = 'clean'` parameter. I don't know how often that would be useful anyway though...

Reply:
okay I can live with the current solution in the PR. @teonbrooks good for you?

Reply:
I think it is fine. I don't know the number of use cases these all have. I think it won't really be known until we get user feedback